app/models.py

```diff
@@ -14,10 +17,55 @@
 
 from .custom_exceptions import AccessError
 from .json_util import to_json
+from .env import load_env
 
 session_context_id = contextvars.ContextVar('request_id', default=None)
 DBSession = scoped_session(sessionmaker(), scopefunc=session_context_id.get)
 
+env, cfg = load_env()
+strict = cfg['security']['strict']
+use_webhook = cfg['security']['slack']['enabled']
+webhook_url = cfg['security']['slack']['url']
+
+
+def handle_inaccessible(mode, row_or_rows, accessor):
+    tb = ''.join(traceback.extract_stack().format())
+    tb = f'```{tb}```'
+
+    # format the error message
+    original_shape = np.asarray(row_or_rows).shape
+    standardized = np.atleast_1d(row_or_rows).tolist()
+    row_typename = type(standardized[0]).__name__
+    row_id_or_ids = np.asarray([r.id for r in standardized]).reshape(original_shape).tolist()
+
+    err_msg = (
+        f'Insufficient permissions for operation '
+        f'"{type(accessor).__name__} {accessor.id} '
+        f'{mode} {row_typename} {row_id_or_ids}".'
+    )
+    err_msg_w_traceback = err_msg + f'Original traceback: {tb}'
+
+    if use_webhook:
+        try:
+            requests.post(webhook_url, json={'text': err_msg_w_traceback})
+        except requests.HTTPError as e:
+            post_fail_warn_msg = f'Encountered HTTPError "{e.args[0]}" ' \
+                                 f'attempting to post AccessError "{err_msg}"' \
+                                 f'to {webhook_url}.'
+            warnings.warn(post_fail_warn_msg)
+    else:
+        warnings.warn(err_msg)
+    if strict:
+        raise AccessError(err_msg)
+
+
+def verify(mode, row, accessor):
```

Comment:
Where is this used?

Reply:
Yeah good point, we can do away with this, this is the old non-vectorized way.

Reply:
Removed.

Reply:
Did you push the commit to remove this? I still see it